tests/foreman/cli/test_registration.py

```diff
@@ -224,3 +226,48 @@ def test_negative_global_registration_without_ak(module_target_sat):
         'Failed to generate registration command:\n  Missing activation key!'
         in context.value.message
     )
+
+
+def test_positive_custom_facts_for_host_registration(
+    module_sca_manifest_org,
+    module_location,
+    module_target_sat,
+    rhel8_contenthost,
+    module_activation_key,
+):
+    """Attempt to register a host and check all the interfaces are created from the custom facts
+
+    :id: db73c146-4557-4bf4-a8e2-950ecba31620
+
+    :steps:
+        1. Register the host.
+        2. Check the host is registered and all the interfaces are created successfully.
+
+    :expectedresults: Host registered successfully with all interfaces created from the custom facts.
+
+    :BZ: 2250397
+
+    :customerscenario: true
+    """
+    interface1 = {'name': gen_string('alphanumeric')}
+    interface2 = {'name': gen_string('alpha'), 'vlan_id': gen_string('numeric')}
+    interface3 = {'name': gen_string('alpha'), 'vlan_id': gen_string('numeric')}
```

Comment:
Will suggest to use list here instead of defining individual variable
```suggestion
    interfaces = [
           {'name': gen_string('alphanumeric')},
           {'name': gen_string('alpha'), 'vlan_id': gen_string('numeric')},
           {'name': gen_string('alpha'), 'vlan_id': gen_string('numeric')},
    ]
```